notebooks/02-ingestion/05-partitioning-data.ipynb

```diff
@@ -0,0 +1,717 @@
+    "  \"Cancelled\", \"CancellationCode\", \"Diverted\", \"CRSElapsedTime\", \"ActualElapsedTime\", \"AirTime\", \"Flights\", \"Distance\", \"DistanceGroup\", \"CarrierDelay\",\n",
+    "  \"WeatherDelay\", \"NASDelay\", \"SecurityDelay\", \"LateAircraftDelay\", \"FirstDepTime\", \"TotalAddGTime\", \"LongestAddGTime\", \"DivAirportLandings\", \"DivReachedDest\",\n",
+    "  \"DivActualElapsedTime\",  \"DivArrDelay\",  \"DivDistance\",  \"Div1Airport\",  \"Div1AirportID\",  \"Div1AirportSeqID\",  \"Div1WheelsOn\",  \"Div1TotalGTime\",\n",
+    "  \"Div1LongestGTime\",  \"Div1WheelsOff\",  \"Div1TailNum\",  \"Div2Airport\",  \"Div2AirportID\",  \"Div2AirportSeqID\",  \"Div2WheelsOn\",  \"Div2TotalGTime\",\n",
+    "  \"Div2LongestGTime\",  \"Div2WheelsOff\",  \"Div2TailNum\",  \"Div3Airport\",  \"Div3AirportID\",  \"Div3AirportSeqID\",  \"Div3WheelsOn\",  \"Div3TotalGTime\",\n",
+    "  \"Div3LongestGTime\",  \"Div3WheelsOff\",  \"Div3TailNum\",  \"Div4Airport\",  \"Div4AirportID\",  \"Div4AirportSeqID\",  \"Div4WheelsOn\",  \"Div4TotalGTime\",\n",
+    "  \"Div4LongestGTime\",  \"Div4WheelsOff\",  \"Div4TailNum\",  \"Div5Airport\",  \"Div5AirportID\",  \"Div5AirportSeqID\",  \"Div5WheelsOn\",  \"Div5TotalGTime\",\n",
+    "  \"Div5LongestGTime\",  \"Div5WheelsOff\",  \"Div5TailNum\"\n",
+    "FROM \"ext\"\n",
+    "PARTITIONED BY YEAR\n",
+    "CLUSTERED BY IATA_CODE_Reporting_Airline\n",
+    "'''\n",
+    "# use a request so that we can specify the query context\n",
+    "req = sql_client.sql_request(sql)\n",
+    "req.add_context(\"rowsPerSegment\", \"50000\")  # we still want to target our ideal of 50k per segment, normally greater than 3 million \n",
+    "display.run_task(req)\n",
+    "sql_client.wait_until_ready('flights_year_IATA')"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "8799d234-c2f3-4277-9661-8931b94d58ff",
+   "metadata": {},
+   "source": [
+    "The [segment view for this table](http://localhost:8888/unified-console.html#segments/datasource~flights_year_IATA) shows 12 segments that are still very close to our ideal segment size, but now they have some new metadata based on the clustering columns in the `Shard Spec` column. It shows which range of values of the IATA_CODE_Reporting_Airline column are available in the segment. This is very useful for pruning when filtering on that column because Druid can prune to 1 or 2 segment files when looking for a single airline.\n",
+    "\n",
+    "![](assets/segments-iata.png)\n",
+    "\n",
+    "## Query Performance\n",
+    "Now that the data has been loaded using a few different partitioning strategies, you can use a test query to determine which one is the best.\n",
+    "In order to control the experiment we will turn cacheing off for the results of the query and measure the same query with each table.\n",
```

Comment:
"The useCache" query parameter will be set to false for the purposes of this part of the notebook so that eggs, cheese, and ham." + link to the query params docs page.

Reply:
done